service/resource/configmap/configmap.go

```diff
@@ -1,63 +1,73 @@
 package configmap
 
 import (
-	"context"
+	"k8s.io/client-go/kubernetes"
 
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
 const (
 	Name = "configmap"
 )
 
-type Config struct{}
+type Config struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
 
-func DefaultConfig() Config {
-	return Config{}
-}
-
-type Resource struct{}
-
-func New(config Config) (*Resource, error) {
-	return &Resource{}, nil
+	ConfigMapNamespace string
+	ConfigMapName      string
 }
 
-func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
```

Comment:
Ninja'd these into separate files, to help my brain. Main meat here is `GetCurrentState`